Update to Zig 0.8.0

leroycep opened the pull request:
Some of the major changes this version:

- HashMap iterator has been changed. I fixed these error in the first commit.
- Testing has switched from panicking to gathering a list of error values. Updating to this will mostly require adding `try` in front of `std.testing.expectX`, but it will also require any tester functions to be updated to return error codes. This has not been fixed in this pull request at the moment.

Nomy1 commented on src/ecs/registry.zig:
```diff
@@ -216,7 +216,7 @@ pub const Registry = struct {
     pub fn assure(self: *Registry, comptime T: type) *Storage(T) {
         var type_id = utils.typeId(T);
         if (self.components.getEntry(type_id)) |kv| {
-            return @intToPtr(*Storage(T), kv.value);
```
@leroycep If you haven't found it yet, there exists the same needed update in the dispatcher.zig file as well on line 21
```var signal = @intToPtr(*Signal(void), ptr.value_ptr.*);```

leroycep replied:
Thanks! I've gone ahead and fixed this in the latest commits

leroycep commented:
Alright, I added two commits. The first one switches the tests to return errors, though there were a few cases where I didn't know how to return an error from the function so I just added `catch unreachable`. For example sinks didn't allow returning an error.

The second commit makes the code compatible with an even newer version zig, where unused variables and redundant `comptime` keywords are disallowed.

This accomplishes what I wanted to do with this PR, so I'm opening it up for review.

prime31 commented:
This is looking solid! Awesome job.
